src/web/wms/src/main/java/org/geoserver/wms/web/publish/WMSLayerConfig.java

```diff
@@ -203,12 +204,20 @@ private void initWMSCascadedUI(IModel<LayerInfo> layerModel) {
 
         WMSLayerInfo wmsLayerInfo = (WMSLayerInfo) layerModel.getObject().getResource();
         // for new only
-        if (layerModel.getObject().getId() == null) wmsLayerInfo.reset();
-        else {
-            wmsLayerInfo.getAllAvailableRemoteStyles().clear();
+        try {
+            if (layerModel.getObject().getId() == null) wmsLayerInfo.reset();
+            else {
+                // pull latest styles from remote WMS
+                wmsLayerInfo.getAllAvailableRemoteStyles().clear();
+                wmsLayerInfo
+                        .getAllAvailableRemoteStyles()
+                        .addAll(wmsLayerInfo.getRemoteStyleInfos());
+            }
+        } catch (Exception e) {
+            error("unable to fetch remote styles for " + wmsLayerInfo.getNativeName());
+            LOGGER.severe("unable to fetch remote styles for " + wmsLayerInfo.getNativeName());
+            LOGGER.log(Level.SEVERE, e.getMessage(), e);
```

Comment:
Will this not log the same thing three times? suggestion:
```
catch (Exception exception) {
  error("Unable to fetch remote styles for layer '" + wmsLayerInfo.getNativeName() + "': " + exception.getMessage());
  LOGGER.severe("unable to fetch remote styles for " + wmsLayerInfo.getNativeName());
  LOGGER.log(Level.SEVERE, "Unable to fetch remote styles for layer '" + wmsLayerInfo.getNativeName() + "'.", exception);
```

Reply:
indeed, will remove the middle line, since the first line is printing error message on page